### PR TITLE
test s3-copy-via-encryption 

### PR DESCRIPTION
**Pull request by viduship**
This PR addresses
 s3-copy-via-encryption (bug https://bugzilla.redhat.com/show_bug.cgi?id=2149450) 
    https://polarion.engineering.redhat.com/polarion/#/project/https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83575711

note: the s3 copy test is expected to fail since not been implemented/fixed

### CODE REVIEW

File: rgw/v2/tests/s3_swift/test_compression_encryption.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
+"""
+test_compression_encryption.py - using boto
+
+Usage: test_compression_encryption.py -c <input_yaml>
+<input_yaml>
+    test_compression_encryption.yaml
```
Comment (hmaheswa) [inline rgw/v2/tests/s3_swift/test_compression_encryption.py:6]: @viduship , please add both configs in this PR

File: rgw/v2/tests/s3_swift/test_compression_encryption.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+                                reusable.upload_object(
+                                    s3_object_name,
+                                    bucket,
+                                    TEST_DATA_PATH,
+                                    config,
+                                    each_user,
+                                )
+
+    if config.test_ops["server_side_copy"] is True:
+        bucket1, bucket2 = buckets_created
```
Comment (ckulal) [inline rgw/v2/tests/s3_swift/test_compression_encryption.py:116]: Hi @viduship , what we are doing this line?

**Review by ckulal — COMMENTED** [review]
lgtm

File: rgw/v2/tests/s3_swift/test_compression_encryption.py (hunk 1 above)
Comment (viduship) [reply rgw/v2/tests/s3_swift/test_compression_encryption.py:6]: thanks @hmaheswa , I have updated the second yaml as well

File: rgw/v2/tests/s3_swift/multisite_configs/test_server_side_copy_object_via_encryption.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# polarion test case id: CEPH-83575711
+# bug-2149450
+config:
+     user_count: 1
+     haproxy: true
+     bucket_count: 2
+     objects_count: 10
```
Comment (hmaheswa) [inline rgw/v2/tests/s3_swift/multisite_configs/test_server_side_copy_object_via_encryption.yaml:7]: @viduship , please increase object count

Comment (viduship) [reply rgw/v2/tests/s3_swift/multisite_configs/test_server_side_copy_object_via_encryption.yaml]: addressed

**Review by hmaheswa — DISMISSED** [review]

**Review by ckulal — DISMISSED** [review]

File: rgw/v2/tests/s3_swift/test_compression_encryption.py [hunk 4]
```diff
@@ -0,0 +1,209 @@
+"""
+test_compression_encryption.py - using boto
+
+Usage: test_compression_encryption.py -c <input_yaml>
+<input_yaml>
+    test_compression_encryption.yaml
+    test_s3_copy_compression_encryption.yaml
```
Comment (mkasturi18) [inline rgw/v2/tests/s3_swift/test_compression_encryption.py:7]: @viduship , could you update the details with the conf files

Comment (viduship) [reply rgw/v2/tests/s3_swift/test_compression_encryption.py]: updated @mkasturi18

File: rgw/v2/tests/s3_swift/test_compression_encryption.py [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+        # check for object existence in bucket2
+        for copy_object_name in copy_objects:
+            if copy_object_name in all_bucket2_objs:
+                log.info("server side copy successful")
+            else:
+                raise TestExecError("server side copy operation was not successful")
+
+    if config.test_ops["remote_object_download"] is True:
+        each_user = user1
```
Comment (hmaheswa) [inline rgw/v2/tests/s3_swift/test_compression_encryption.py:153]: @viduship , I think user1 in this line is not defined, could you please check once

Comment (viduship) [reply rgw/v2/tests/s3_swift/test_compression_encryption.py]: hi @hmaheswa , I have removed the remote download case as of now and only kept the s3 copy one.

Please review

**Review by hmaheswa — APPROVED** [review]
lgtm

**Review by mkasturi18 — APPROVED** [review]

**Pull request merged**